src/ToeplitzMatrices.jl

```diff
@@ -29,6 +30,7 @@ convert(::Type{AbstractToeplitz{T}}, A::AbstractToeplitz) where T = AbstractToep
 
 isconcrete(A::AbstractToeplitz) = isconcretetype(typeof(A.vc)) && isconcretetype(typeof(A.vr))
 iszero(A::AbstractToeplitz) = iszero(A.vc) && iszero(A.vr)
+diag(A::AbstractToeplitz) = Fill(A.vc[1], min(size(A)...))
```

Comment:
This can be done as a `fill`, right? That way, the dependency on `FillArrays` isn't needed (although it's not a bad idea in itself)

Reply:
As of Julia 1.9, how about using `fill` as the minimal feature and override it with `Fill` in the extension?

Reply:
We wouldn't want to override a method, that'll hurt precompilation. Extensions are for adding extra methods.

That said, perhaps using `FillArrays` isn't a bad idea after all

Reply:
If both `vc` and `vr` are lazy and large then using `fill` could cause a problem. 

`range(x, x, length=n)` could mimic `Fill(x, n)` though.